project/src/main/webapp/emailParse.js

```diff
@@ -11,92 +11,87 @@
+        const question = gmail.question;
+        const totalObjects = gmail.totalObjects;
+        const model = this.model;
+        const allPass = this.extractEmailBodiestoArray(totalObjects);
+        console.time("Using Promises Test");
+        const promises = allPass.map(
+            passage => model.findAnswers(question, passage),
+        );
+        Promise.all(promises).then(values => {
+            const nonEmpty = this.ridOfEmptyArrays(values, allPass);
+            const answer = document.getElementById("answer");
+            if (nonEmpty.size == 0) {
+                answer.innerHTML = "No Answer Available";
+            }
+            else {
+                const orderedConfidence = this.highestConfidence(nonEmpty);
+                const mlDictAnswer = nonEmpty.get(orderedConfidence);
+                console.log("original", values);
+                console.log("nonEmpty", nonEmpty);
+                console.log("highest confidence", orderedConfidence);
+                answer.innerHTML = mlDictAnswer["answer"];
+                console.timeEnd("Using Promises Test");
+            }
+        });
+    }
 
-function parseEmailsWithModel()
-{
-    var question = "When is my Microsoft Interview?"
-    var exampleDict = {
-        "12345":{"date": "Some string", "body": "Your Microsoft interview will be held on October 2, 2019 at 01:45 PM Pacific Time (US & Canada)"},
-        "23456":{"date": "Some string", "body": "Arely Miranda González, a Product Manager at YouTube, started Karma Action, a new non-profit organization and webapp"},
-        "34567":{"date": "Some string", "body": "Your microsoft workshop is on Monday 4, 2019."},
-    };
-    var allPass = listOfPassages(exampleDict);
-    console.time("Using Promises Test");
-    const promises =allPass.map(
-        passage => model.findAnswers(question, passage),
-    );
-    Promise.all(promises).then((values) => {
-        var nonEmpty = ridOfEmptyArrays(values,allPass);
-        if (nonEmpty.size == 0)
-        {
-            console.log("No Answer Available");
+    extractEmailBodiestoArray(exampleDict) {
+        let allPass = [];
+        for (let key in exampleDict) {
+            let body = exampleDict[key]["emailBody"];
+            allPass.push(body.slice(300));
         }
-        else
-        {
-            var orderedConfidence = highestConfidence(nonEmpty);
-            var MLDictAnswer = nonEmpty.get(orderedConfidence);
-            console.log("original",values);
-            console.log("nonEmpty", nonEmpty);
-            console.log("highest confidence", orderedConfidence);
-            console.log("final answer: ", MLDictAnswer["answer"]);
-            var answer = document.getElementById("answer");
-            answer.innerHTML = MLDictAnswer["answer"];
-            console.timeEnd("Using Promises Test");
+        return allPass;
+    }
+    /** Keep emails that returned an answer from ML model */
+    ridOfEmptyArrays(mlValues, allPass) {
```

Comment:
I think this method name and jsdoc is a little misleading. Maybe something like `getScoreToEmail`?